src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -818,6 +825,16 @@ private void Parse()
                     "Specifying both tlsInsecure and tlsDisableCertificateRevocationCheck is invalid.");
             }
 
+            if (_scheme == ConnectionStringScheme.MongoDBPlusSrv && _connect == ClusterConnectionMode.Direct)
+            {
+                throw new MongoConfigurationException("Direct connect is not allowed to be used with SRV.");
+            }
+
+            if (_hosts.Count > 1 && _connect == ClusterConnectionMode.Direct)
+            {
+                throw new MongoConfigurationException("Direct connect is not allowed to be used with multiple seeds.");
```

Comment:
Suggested error message change:

    throw new MongoConfigurationException("Direct connect cannot be used with multiple host names.");